lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -227,74 +194,144 @@ Point TruncatedDistribution::getRealization() const
   for (;;)
   {
     Point realization(distribution_.getRealization());
-    if ((!finiteLowerBound_ || realization[0] >= lowerBound_) && (!finiteUpperBound_ || realization[0] <= upperBound_)) return realization;
+    if (bounds_.contains(realization)) return realization;
```

Comment:
There are several problems here:
  + The CDF inversion works only for 1D distributions (and for any distribution, not only continuous ones)
  + The test must be done wrt the probability of the bounding interval according to the underlying non-truncated distribution. As this probability is not stored but its inverse, the test becomes normalizationFactor_ * thresholdRealization_ > 1.0
  + line 196 Point is const
  + The multivariate counterpart of CDF inversion is InverseRosenblattEvaluation feeded with DistFunc::rNormal(getDimension()). In fact, Rosenblatt/InverseRosenblatt should map to or from the uniform distribution on the unit cube instead of the Gaussian standard space and DistFunc::rNormal(getDimension()) should be replaced by RandomGenerator::Generate(getDimension())
  + If one uses the InverseRosenblattEvaluation class, one should implement the computeConditionalQuantile() and computeConditionalCDF() methods. I don't know if it is easy or not to get these methods as simple modifications of the non-truncated counterparts.